pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/extensions/channel/ServiceUnitStateChannelImpl.java

```diff
@@ -522,12 +590,22 @@ private void handleAssignEvent(String serviceUnit, ServiceUnitStateData data) {
     }
 
     private void handleReleaseEvent(String serviceUnit, ServiceUnitStateData data) {
-        if (isTargetBroker(data.sourceBroker())) {
-            ServiceUnitStateData next = new ServiceUnitStateData(Owned, data.broker(), data.sourceBroker());
-            // TODO: when close, pass message to clients to connect to the new broker
-            closeServiceUnit(serviceUnit)
-                    .thenCompose(__ -> pubAsync(serviceUnit, next))
-                    .whenComplete((__, e) -> log(e, serviceUnit, data, next));
+
+        if (isTransferCommand(data)) {
+            if (isTargetBroker(data.sourceBroker())) {
+                ServiceUnitStateData next = new ServiceUnitStateData(Owned, data.broker(), data.sourceBroker());
+                // TODO: when close, pass message to clients to connect to the new broker
+                closeServiceUnit(serviceUnit)
+                        .thenCompose(__ -> pubAsync(serviceUnit, next))
+                        .whenComplete((__, e) -> log(e, serviceUnit, data, next));
+            }
+        } else {
+            if (isTargetBroker(data.broker())) {
+                ServiceUnitStateData next = new ServiceUnitStateData(Free, data.broker());
+                closeServiceUnit(serviceUnit)
```

Comment:
Since we already closed the service unit here, do we need to close the service unit again in `handleInitEvent`?